eduroam.md

```diff
@@ -40,98 +53,190 @@ The process of connecting to the network may be different depending on the Linux
+- Entering `TTLS` instead of `TLS`. This results in a `journalctl` error shown below:
+```
+EAP server tried method 13 while client was configured for method 21
+EAP completed with eapFail
+4-Way handshake failed for ifindex 4, reason: 23
 ```
-The first of these commands will ask for a password. Enter the password of the file.
 
-3. Now that these files are decrypted, we can create the configuration file for `wpa_supplicant`. Open your text editor of choice and create a file in `/etc/wpa_supplicant`, and fill in the file according to the block below. You can name the file whatever you wish, but for automation purposes, its recommended to name the file `wpa_supplicant-[interface name].conf`. This makes it easier for `systemd` and `dhcpcd` to interface with `wpa_supplicant` once it's set up.
+### wpa_supplicant Configuration
+
+If you do not have a `.p12` file from RIT before starting this section, go back to the beginning of this page, and create a new certificate, downloading the `P12` option instead.
+
+This tutorial assumes that all commands are either run directly as root.
+
+1. Run the following command to determine what network devices are available. This tutorial assumes the use of the network interface `wlan0`.
+```
+# wpa_cli interface
+```
+2. Move both the RIT CA Cert and the encrypted `.p12` file into a common directory that you don't plan on interacting with much. For testing purposes, `/opt/` was used. Locations within `~` or `/home/$USER` may not work properly, due to improper permissions, although this is untested as of writing.
+3. Open your text editor of choice and create a file in `/etc/wpa_supplicant`, and fill in the file according to the block below. You can name the file whatever you wish, but for automation purposes, its recommended to name the file `wpa_supplicant-[interface name].conf`. This makes it easier for `systemd` and `dhcpcd` to interface with `wpa_supplicant` once it's set up.
 ```
 ctrl_interface=/var/run/wpa_supplicant
 ctrl_interface_group=wheel
 update_config=1
+
 network={
     ssid="eduroam"
     scan_ssid=1
     key_mgmt=WPA-EAP
+    proto=WPA2
+    pairwise=CCMP
+    group=CCMP
     eap=TLS
+    anonymous_identity="anonymous@rit.edu"
+    domain_match="radius.rit.edu"
     identity="abc1234@rit.edu"
     ca_cert="/location/of/RIT/cert"
-    client_cert="location/of/decryptedCert.pem"
-    private_key="location/of/decryptedKey.pem"
+    private_key="location/of/p12PrivateKey.p12"
+    private_key_passwd="P@ssw0rd1"
 }
 ```
 
-4. You're all set! Run the following commands in a terminal to start `wpa_supplicant`, then start `dhcpcd`. After this, your network should be set up properly. 
-```bash
-sudo wpa_supplicant -i [interface name] -c /etc/wpa_supplicant/[config file name]
-sudo dhcpcd [interface name]
+4. You're all set! Run the following commands in a terminal to start `wpa_supplicant`, then start `dhcpcd`. After this, your network should be set up properly.
```

Comment:
it might be helpful to also summarize what the commands do or include a link to explainshell.com to help people learn in case they are using linux for the first time. "trust me, run these commands" doesnt seem like its the best habit to be encouraging new linux users to get into.

Reply:
Will edit to expand on command actions.

Reply:
Unfortunately, `explainshell.com` doesn't contain any information on `wpa_supplicant`, and its information on `dhcpcd` is... less than helpful.